moveit_core/collision_detection/src/world.cpp

```diff
@@ -136,6 +136,95 @@ bool World::hasObject(const std::string& object_id) const
   return objects_.find(object_id) != objects_.end();
 }
 
+bool World::knowsTransform(const std::string& object_id) const
+{
+  // Check object names first
+  if (objects_.find(object_id) != objects_.end())
+    return true;
+  else  // Then objects' subframes
+  {
+    if (object_id.find("/")!=std::string::npos)
+    {
+      for (const std::pair<std::string, ObjectPtr>& object : objects_)
+      {
+        try 
+        {
+          // Compare both object name and subframe name
+          int name_length = object.second->id_.length();
+          if (object.second->id_ == object_id.substr(0,name_length))
+          {
+            if (object.second->subframe_poses_.find(object_id.substr(name_length+1)) != object.second->subframe_poses_.end())
+              return true;
+          }
+        }
+        catch (const std::exception& e) {;}
```

Comment:
What kind of exception do you want to catch here?

Reply:
The object name can be longer than the subframe name ("bit/tip" vs "cylinder"), so `substr` indexes out of range.

Reply:
Ah. My solution, using `boost::starts_with` handles this gracefully as well.